src/runtime/d3d12compute.cpp

```diff
@@ -0,0 +1,2794 @@
+        }
+        lib = d3d12_load_library(lib_names[i]);
+    }
+
+    #if HALIDE_D3D12_RENDERDOC
+        #if !RENDERDOC_AUTOINIT
+            TRACEPRINT("Initializing RenderDoc\n");
+            bool rdinit = InitRenderDoc();
+            halide_assert(user_context, rdinit);
+        #endif
+    #endif
+
+    D3D12CreateDevice           = LibrarySymbol::get(user_context, lib_d3d12,           "D3D12CreateDevice");
+    D3D12GetDebugInterface      = LibrarySymbol::get(user_context, lib_d3d12,           "D3D12GetDebugInterface");
+    D3D12SerializeRootSignature = LibrarySymbol::get(user_context, lib_d3d12,           "D3D12SerializeRootSignature");
+    D3DCompile                  = LibrarySymbol::get(user_context, lib_D3DCompiler_47,  "D3DCompile");
+    CreateDXGIFactory1          = LibrarySymbol::get(user_context, lib_dxgi,            "CreateDXGIFactory1");
+
+    // Windows x64 follows the LLP64 integer type convention:
+    // https://msdn.microsoft.com/en-us/library/windows/desktop/aa383751(v=vs.85).aspx
+    halide_assert(user_context, sizeof(BOOL)     == ( 32 / 8)); // BOOL      must be  32 bits
+    halide_assert(user_context, sizeof(CHAR)     == (  8 / 8)); // CHAR      must be   8 bits
+    halide_assert(user_context, sizeof(SHORT)    == ( 16 / 8)); // SHORT     must be  16 bits
+    halide_assert(user_context, sizeof(LONG)     == ( 32 / 8)); // LONG      must be  32 bits
+    halide_assert(user_context, sizeof(ULONG)    == ( 32 / 8)); // ULONG     must be  32 bits
+    halide_assert(user_context, sizeof(LONGLONG) == ( 64 / 8)); // LONGLONG  must be  16 bits
+    halide_assert(user_context, sizeof(BYTE)     == (  8 / 8)); // BYTE      must be   8 bits
+    halide_assert(user_context, sizeof(WORD)     == ( 16 / 8)); // WORD      must be  16 bits
+    halide_assert(user_context, sizeof(DWORD)    == ( 32 / 8)); // DWORD     must be  32 bits
+    halide_assert(user_context, sizeof(WCHAR)    == ( 16 / 8)); // WCHAR     must be  16 bits
+    halide_assert(user_context, sizeof(INT)      == ( 32 / 8)); // INT       must be  32 bits
+    halide_assert(user_context, sizeof(UINT)     == ( 32 / 8)); // UINT      must be  32 bits
+    halide_assert(user_context, sizeof(IID)      == (128 / 8)); // COM GUIDs must be 128 bits
+
+    // Paranoid checks (I am not taking any chances...)
+    halide_assert(user_context, sizeof(INT8)   == ( 8 / 8));
+    halide_assert(user_context, sizeof(INT16)  == (16 / 8));
+    halide_assert(user_context, sizeof(INT32)  == (32 / 8));
+    halide_assert(user_context, sizeof(INT64)  == (64 / 8));
+    halide_assert(user_context, sizeof(UINT8)  == ( 8 / 8));
+    halide_assert(user_context, sizeof(UINT16) == (16 / 8));
+    halide_assert(user_context, sizeof(UINT32) == (32 / 8));
+    halide_assert(user_context, sizeof(UINT64) == (64 / 8));
+#ifdef BITS_64
+    halide_assert(user_context, sizeof(SIZE_T) == (64 / 8));
+#else
+    halide_assert(user_context, sizeof(SIZE_T) == (32 / 8));
+#endif
+}
+
+#if HALIDE_D3D12_PIX
```

Comment:
What's PIX/what's this used for?

Reply:
PIX is a D3D graphics debugger. It's more in-depth than RenderDoc (like NVIDIA NSight).
In the desktop arena, PIX was popular over a decade ago, but got neglected and only evolved in the console front (namely, XBox360 and XBoxOne). Now PIX for Windows is back.